src/test/test-varlink.c

```diff
@@ -300,13 +301,24 @@ int main(int argc, char *argv[]) {
         assert_se(varlink_server_attach_event(s, e, 0) >= 0);
         assert_se(varlink_server_set_connections_max(s, OVERLOAD_CONNECTIONS) >= 0);
 
+        assert_se(json_build(&v, JSON_BUILD_OBJECT(JSON_BUILD_PAIR("a", JSON_BUILD_INTEGER(7)),
+                                                   JSON_BUILD_PAIR("b", JSON_BUILD_INTEGER(22)))) >= 0);
+
+        assert_se(varlink_connect_address(&d, sp) >= 0);
+        assert_se(varlink_set_description(d, "collect-client") >= 0);
+
+        assert_se(varlink_collect(d, "io.test.DoSomething", v, &o) >= 0);
```

Comment:
We should add a new server method that uses `VARLINK_MORE` and test against that instead so we exercise the functionality of varlink_collect() properly.